sdk/core/Azure.Core.Amqp/tests/AmqpAnnotatedMessageConverterTests.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework;
+
+namespace Azure.Core.Amqp.Tests
+{
+    public class AmqpAnnotatedMessageConverterTests
+    {
+        private static readonly object[] s_amqpValues =
+        {
+            "string",
+            new List<string> {"first", "second"},
+            'c',
+            5,
+            new int[] { 5 },
+            long.MaxValue,
+            new long[] { long.MaxValue },
+            (byte) 1,
+            (sbyte) 1,
+            (short) 1,
+            (ushort) 1,
+            3.1415926,
+            new double[] { 3.1415926 },
+            new decimal(3.1415926),
+            new decimal[] { new decimal(3.1415926) },
+            DateTimeOffset.Parse("3/24/21").UtcDateTime,
+            new DateTime[] {DateTimeOffset.Parse("3/24/21").UtcDateTime },
+            DateTimeOffset.Parse("3/24/21"),
+            new DateTimeOffset[] {DateTimeOffset.Parse("3/24/21") },
+            TimeSpan.FromSeconds(5),
+            new TimeSpan[] {TimeSpan.FromSeconds(5)},
+            new Uri("http://localHost"),
+            new Uri[] { new Uri("http://localHost") },
+            new Guid("55f239a6-5d50-4f6d-8f84-deed326e4554"),
+            new Guid[] { new Guid("55f239a6-5d50-4f6d-8f84-deed326e4554"), new Guid("55f239a6-5d50-4f6d-8f84-deed326e4554") },
+            new Dictionary<string, string> { { "key", "value" } },
+            new Dictionary<string, char> {{ "key", 'c' }},
+            new Dictionary<string, int> {{ "key", 5 }},
+            new Dictionary<string, byte> {{ "key", 1 } },
+            new Dictionary<string, sbyte> {{ "key", 1 } },
+            new Dictionary<string, short> {{ "key", 1 } },
+            new Dictionary<string, double> {{ "key", 3.1415926 } },
+            new Dictionary<string, decimal> {{ "key", new decimal(3.1415926) } },
+            new Dictionary<string, DateTime> {{ "key", DateTimeOffset.Parse("3/24/21").UtcDateTime } },
+            // for some reason dictionaries with DateTimeOffset, Timespan, or Uri values are not supported in AMQP lib
```

Comment:
They need to be set as described types, I believe.  You can't use them as just a plain value type.

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/AmqpMessageConverterTests.cs#L49

Reply:
Interesting.. Should we be handling this in the AmqpAnnoatedMessageConverter?

Reply:
This would only impact value-body sections which are not widely used.

Reply:
> Should we be handling this in the AmqpAnnotatedMessageConverter? 

Maybe?  We convert for direct properties in maps.   As you noted, we do not crack open nested values nor do we do the Value body.    I don't think we want to start poking at nested values, but the value body may be nice.   

That said, it's such a specialized niche case that it's probably not worth the investment.

Reply:
Agreed.